cmd/msrc/generate.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/server/vulnerabilities/msrc"
+	"github.com/fleetdm/fleet/v4/server/vulnerabilities/msrc/io"
+	"github.com/fleetdm/fleet/v4/server/vulnerabilities/msrc/parsed"
+	"github.com/google/go-github/v37/github"
+)
+
+func panicif(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+func main() {
+	wd, err := os.Getwd()
+	panicif(err)
+
+	now := time.Now()
+	httpC := http.DefaultClient
+
+	ghAPI := io.NewGithubClient(httpC, github.NewClient(httpC).Repositories, wd)
+	msrcAPI := io.NewMSRCClient(httpC, wd, nil)
+
+	fmt.Println("Downloading existing bulletins...")
+	eBulletins, err := ghAPI.Bulletins()
+	panicif(err)
+
+	var bulletins []*parsed.SecurityBulletin
+	if len(eBulletins) == 0 {
+		fmt.Println("None found, backfilling...")
+		bulletins, err = backfill(now.Month(), now.Year(), msrcAPI)
+		panicif(err)
+	} else {
+		fmt.Println("Updating existing bulletins")
+		bulletins, err = update(now.Month(), now.Year(), eBulletins, msrcAPI, ghAPI)
+		panicif(err)
+	}
+
+	fmt.Println("Saving bulletins...")
+	for _, b := range bulletins {
+		err := serialize(b, now, wd)
+		panicif(err)
+	}
+
+	fmt.Println("Done.")
+}
+
+func update(
```

Comment:
should `backfill` and `update` just receive a `time.Time`? will allow them higher granularity if needed in the future and simplifies the API

Reply:
The query params for the  MSRC API only uses month and year - maybe is a matter of personal taste - but for me it would be a little confusing to see that we pass in a `Time` when we only use the month and year - but if it makes the code easier to read, I'm onboard.

Reply:
makes sense to keep them as-is, thanks for explaining!